js/app/dashboard/store/reducers.js

```diff
@@ -251,12 +253,54 @@ const taskListGroupMode = (state = 'GROUP_MODE_FOLDERS', action) => {
   }
 }
 
+const tasksFilters = (state = {showFinishedTasks: true}, action) => {
+  switch (action.type) {
+    case 'TOGGLE_SHOW_FINISHED_TASKS':
+      let showFinishedTasks = !state.showFinishedTasks
+      return {...state, showFinishedTasks}
+    default:
+      return state
+  }
+}
+
+const allTags = (state = tagsInitial, action) => {
+  return state
+}
+
+
+const tagsFilters = (state = {selectedTags:  tagsInitial}, action ) => {
```

Comment:
Actually I thought that everything regarding tags will go in the `tasksFilters` reducer. It's ok to have a separate reducer (don't know the best choice), but in this case the state can simply be the list of selectedTags